pinot-common/src/main/java/org/apache/pinot/common/utils/filefetcher/BaseFileFetcher.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.utils.filefetcher;
+
+import java.io.File;
+import java.net.URI;
+import org.apache.commons.configuration.Configuration;
+import org.apache.pinot.spi.utils.retry.RetryPolicies;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Base implementation of file fetcher with the retry logic embedded.
+ */
+public abstract class BaseFileFetcher implements FileFetcher {
+  public static final String RETRY_COUNT_CONFIG_KEY = "retry.count";
+  public static final String RETRY_WAIT_MS_CONFIG_KEY = "retry.wait.ms";
+  public static final int DEFAULT_RETRY_COUNT = 3;
+  public static final int DEFAULT_RETRY_WAIT_MS = 100;
+
+  protected final Logger _logger = LoggerFactory.getLogger(getClass().getSimpleName());
+
+  protected int _retryCount;
+  protected int _retryWaitMs;
+
+  @Override
+  public void init(Configuration config) {
+    _retryCount = config.getInt(RETRY_COUNT_CONFIG_KEY, DEFAULT_RETRY_COUNT);
+    _retryWaitMs = config.getInt(RETRY_WAIT_MS_CONFIG_KEY, DEFAULT_RETRY_WAIT_MS);
+    doInit(config);
+    _logger.info("Initialized with retryCount: {}, retryWaitMs: {}", _retryCount, _retryWaitMs);
+  }
+
+  /**
+   * Override this for custom initialization.
+   */
+  protected void doInit(Configuration config) {
+  }
+
+  @Override
+  public void fetchFileToLocal(URI uri, File dest)
+      throws Exception {
+    RetryPolicies.fixedDelayRetryPolicy(_retryCount, _retryWaitMs).attempt(() -> {
+      try {
+        fetchFileToLocalWithoutRetry(uri, dest);
+        _logger.info("Fetched file from URI: {} to local: {} of size: {}", uri, dest, dest.length());
+        return true;
+      } catch (Exception e) {
+        _logger.warn("Caught exception while fetching file from URI: {} to local file: {}, will retry", uri, dest, e);
```

Comment:
"will retry" is misleading. It is not true if this is the last time we get this exception. Maybe just remove it, since we don't have a a mechanism to find the number of retries in here.

Reply:
Done